library/packages/src/lib/y2packager/zypp_url.rb

```diff
@@ -146,7 +146,10 @@ def inspect
     # object is introduced to replace an existing URI one.
     def ==(other)
       if other.is_a?(URI::Generic)
-        uri == other
+        uri == other ||
+          # zypp will normalize dir:///foo to dir:/foo
+          # https://github.com/openSUSE/libzypp/issues/441
+          uri == URI(other.to_s.sub(":///", ":/"))
```

Comment:
can we use it in generaal ofr all URI types? or only for dir?

Reply:
Short answer: yes

`URI::Generic` is a base class for all the Ruby URIs. The `:///` means there is an empty *authority* part, where *authority* is [user[:password]@]host[:port], and `:/` means the authority is undefined. Ruby 3.2 normalizes it to 3 slashes, zypp normalizes it to 1 slash.

But I may be misunderstanding your question. How can I improve the code or the comment?

Reply:
it zypp do it for *all* triple slash, then it is fine for me. I just want to be sure.